projects/js-packages/boost-score-api/tests/index.test.js

```diff
@@ -1,8 +1,86 @@
-import { requestSpeedScores } from '../src/index';
+import api from '../src/api';
+import {
+	requestSpeedScores,
+	getScoreLetter,
+	didScoresChange,
+	getScoreMovementPercentage,
+} from '../src/index';
+
+const mockData = {
+	status: 'success',
+	timestamp: 123456789,
+	scores: {
+		current: {
+			desktop: 90,
+			mobile: 80,
+		},
+		noBoost: {
+			desktop: 90,
+			mobile: 80,
+		},
+		isStale: true,
+	},
+	theme: '',
+};
 
-// This is a placeholder test, new tests will be added soon
 describe( 'requestSpeedScores', () => {
-	it( 'should be a function', () => {
-		expect( typeof requestSpeedScores ).toBe( 'function' );
+	beforeEach( () => {
+		jest.spyOn( api, 'post' );
+	} );
+
+	afterEach( () => {
+		jest.restoreAllMocks();
+	} );
+
+	it( 'should return speed scores', async () => {
+		api.post.mockResolvedValue( mockData );
+
+		const scores = await requestSpeedScores( 'https://example.com' );
+		expect( scores ).toEqual( mockData.scores );
+	} );
+} );
+
+describe( 'getScoreLetter', () => {
+	it( 'Should return the correct score', () => {
+		expect( getScoreLetter( 90, 91 ) ).toBe( 'A' );
+		expect( getScoreLetter( 90, 83 ) ).toBe( 'B' );
+		expect( getScoreLetter( 90, 60 ) ).toBe( 'C' );
+		expect( getScoreLetter( 45, 50 ) ).toBe( 'D' );
+		expect( getScoreLetter( 26, 30 ) ).toBe( 'E' );
+		expect( getScoreLetter( 0, 0 ) ).toBe( 'F' );
+	} );
+} );
+
+describe( 'didScoresChange', () => {
+	it( 'Should return false if scores did not change', () => {
+		expect( didScoresChange( mockData.scores ) ).toBe( false );
+	} );
+
+	it( 'should return true if scores changed', () => {
+		mockData.scores.noBoost.desktop = 60;
+		mockData.scores.noBoost.mobile = 50;
```

Comment:
Maybe we should create and work on a copy of `mockData` object instead? So as not to modify a global object that we'll reuse in other tests. It might just lead to unexpected errors.
```suggestion
		const changedMockData = Object.assign({}, mockData);
		changedMockData.scores.noBoost.desktop = 60;
		changedMockData.scores.noBoost.mobile = 50;
```

Other option I would see is to define `mockData` as `let` and re-define it in `beforeEach` fn.

Reply:
You're right, this is better. I'll get that updated, thanks!